### ISSUE TITLE
Ulauncher throws an error on launch

### ISSUE THREAD

**Issue by critbase** [NixOS/nixpkgs#119350]
**Describe the bug**
I'm unsure what the error is, but it seems to relate to ulauncher being unable to find the project path, whatever that may be. 

**To Reproduce**
1. `nix-shell -p ulauncher --run "ulauncher"`

**Expected behavior**
For Ulauncher to start normally

**Additional context**
This is the error message I get:
```2021-04-13 13:18:54,780 | ERROR | ulauncher: except_hook() | Uncaught exception
Traceback (most recent call last):
  File "/nix/store/4dw42b5pf1ngkk9icj0xz14f1v6x8gk4-ulauncher-5.9.0/lib/python3.8/site-packages/ulauncher/config.py", line 68, in get_data_path
    return paths[0]
IndexError: list index out of range

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/nix/store/4dw42b5pf1ngkk9icj0xz14f1v6x8gk4-ulauncher-5.9.0/bin/..ulauncher-wrapped-wrapped", line 29, in <module>
    main()
  File "/nix/store/4dw42b5pf1ngkk9icj0xz14f1v6x8gk4-ulauncher-5.9.0/lib/python3.8/site-packages/ulauncher/main.py", line 128, in main
    window = UlauncherWindow.get_instance()
  File "/nix/store/4dw42b5pf1ngkk9icj0xz14f1v6x8gk4-ulauncher-5.9.0/lib/python3.8/site-packages/ulauncher/utils/decorator/singleton.py", line 19, in wrapper
    instance = fn(*args, **kwargs)
  File "/nix/store/4dw42b5pf1ngkk9icj0xz14f1v6x8gk4-ulauncher-5.9.0/lib/python3.8/site-packages/ulauncher/ui/windows/UlauncherWindow.py", line 55, in get_instance
    return cls()
  File "/nix/store/4dw42b5pf1ngkk9icj0xz14f1v6x8gk4-ulauncher-5.9.0/lib/python3.8/site-packages/ulauncher/ui/windows/UlauncherWindow.py", line 63, in __new__
    builder = Builder.new_from_file('UlauncherWindow')
  File "/nix/store/4dw42b5pf1ngkk9icj0xz14f1v6x8gk4-ulauncher-5.9.0/lib/python3.8/site-packages/ulauncher/ui/windows/Builder.py", line 47, in new_from_file
    ui_filename = get_data_file('ui', '%s.ui' % (builder_file_name,))
  File "/nix/store/4dw42b5pf1ngkk9icj0xz14f1v6x8gk4-ulauncher-5.9.0/lib/python3.8/site-packages/ulauncher/config.py", line 42, in get_data_file
    return os.path.join(get_data_path(), *path_segments)
  File "/nix/store/4dw42b5pf1ngkk9icj0xz14f1v6x8gk4-ulauncher-5.9.0/lib/python3.8/site-packages/ulauncher/config.py", line 70, in get_data_path
    raise ProjectPathNotFoundError()
ulauncher.config.ProjectPathNotFoundError
```
**Notify maintainers**

@aaronjanse @worldofpeace 

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.27, NixOS, 21.05.20210410.a73020b (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210326_dd77f71`
 - channels(root): `"nixos-21.05pre274021.d496205cf22, nixpkgs-21.05pre273666.d1257435332"`
 - channels(miyuki): `"home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by aaronjanse** [comment]
See the PR linked above. The issue is that the ulauncher package must be installed via `nix-env`, `nix profile`, `systemPackages`, or something similar. Using `nix-shell` does not work (for reasons explained in the patch). The PR both updates ulauncher and adds a note to explain what's going on to Nix users trying out the package.

I think it would be a good idea to create a NixOS module for Ulauncher. I might be able to do this after I finish exams in a couple weeks.

**Comment by gytis-ivaskevicius** [comment]
Alright guys, the real solution here is to just use rofi. 

With a little work you can make it look pretty good and with a little extra work one could easily make it look similar to ulauncher. Here is my config https://www.github.com/gytis-ivaskevicius/nixfiles/tree/master/overlays%2Fg-rofi

I tried to fix this issue up but it seemed to work for maintainers thus it was the end of that story

**Comment by aaronjanse** [comment]
> it seemed to work for maintainers thus it was the end of that story

Hello @gytis-ivaskevicius, ulauncher package maintainer here. This definitely shouldn't be the end of the story. I do want to get these issues fixed.

The problem is that ulauncher wants to store paths to its assets in a config file within `$HOME`. Typically we'd just let it use `/nix/store` paths for everything, but that would cause ulauncher to break if we update the derivation then garbage-collect the old paths in `/nix/store`.

---

I also use Rofi right now, and it works. But I really want the plugin ecosystem of Ulauncher, so I'm motivated to fix this situation.

Perhaps we could modify the current package to allow decoratively specifying the settings.json file contents (like `vscode-with-extensions`), which would allow the package to be used without messing with system state?

Maybe we could even create something like `ulauncherPackages` and a NixOS module?

**Comment by critbase** [comment]
Perhaps this is a good fit for a home-manager module?
In any case, this issue can probably be closed once #121557 is merged. Maybe a new issue should be opened to track the addition of an unlauncher module?

**Comment by PAI5REECHO** [comment]
bad bot

**Comment by tmarkov** [comment]
I installed `ulauncher` (rev 3a1054b) through `configuration.nix`. Unfortunately, I get this error message on launch anyway.
